engine/src/main/java/org/terasology/engine/world/chunks/localChunkProvider/LocalChunkInitialProvider.java

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.world.chunks.localChunkProvider;
+
+import org.joml.Vector3ic;
+import org.terasology.engine.world.block.BlockRegion;
+import org.terasology.engine.world.chunks.Chunk;
+import org.terasology.engine.world.chunks.internal.ChunkRelevanceRegion;
+import org.terasology.engine.world.chunks.pipeline.InitialChunkProvider;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+public class LocalChunkInitialProvider implements InitialChunkProvider {
+    private final LocalChunkProvider chunkProvider;
+    private final RelevanceSystem relevanceSystem;
+
+    private final List<Vector3ic> chunksInRange;
+    private BlockRegion[] lastRegions;
+
+    public LocalChunkInitialProvider(LocalChunkProvider chunkProvider, RelevanceSystem relevanceSystem) {
+        this.chunkProvider = chunkProvider;
+        this.relevanceSystem = relevanceSystem;
+        chunksInRange = new ArrayList<>();
+        updateList();
+    }
+
+    private void updateList() {
+        relevanceSystem.neededChunks().forEach(chunkPos -> {
+            if (!chunksInRange.contains(chunkPos)) {
+                chunksInRange.add(chunkPos);
+            }
+        });
```

Comment:
can `neededChunks()` contain duplicates? Otherwise, we could sort out the positions already contained in a `filter` step beforehand:

```suggestion
        relevanceSystem.neededChunks()
                .filter(pos -> !chunksInRange.contains(pos))
                .forEach(chunksInRange::add);
```

Reply:
Good idea, that's a lot cleaner.